elyra/cli/pipeline_app.py

```diff
@@ -149,6 +151,42 @@ def _preprocess_pipeline(pipeline_path: str, runtime: str, runtime_config: str)
     return pipeline_definition
 
 
+def _print_issues(issues):
+    # print validation issues
+    for issue in issues:
+        if issue.get('severity') == ValidationSeverity.Error:
+            click.echo(
+                f'- (Fatal error on node \'{issue["data"]["nodeName"]}\' property \'{issue["data"]["propertyName"]}\') '
+                f'- {issue["message"]}')
+        else:
+            # TODO check warning nodeNames are empty
+            click.echo(
+                f'- (Warning) - {issue["message"]}')
+
+    click.echo("")
+
+
+def _validate_pipeline_definition(pipeline_definition):
+    # validate pipeline
+    validation_response = asyncio.get_event_loop().run_until_complete(
+        PipelineValidationManager.instance().validate(pipeline=pipeline_definition))
+
+    if validation_response.has_fatal:
+        click.echo('Pipeline validation FAILED:')
+
+        # print validation issues
+        issues = validation_response.to_json().get('issues')
+        _print_issues(issues)
+
+        raise click.ClickException("Error validating pipeline.")
+    else:
+        click.echo('Pipeline validation WARNINGS:')
+
+        # print validation issues
+        issues = validation_response.to_json().get('issues')
+        _print_issues(issues)
```

Comment:
Seems like this should all be in a general function that takes a single `validation_response` parameter and it raises on fatal errors.

There are also `Information` and `Hint` severity levels besides `Error` and `Warning`, so the else should be more sensitive to that.